lib/cli-engine.js

```diff
@@ -591,6 +635,12 @@ class CLIEngine {
                 results.push(createIgnoreResult(resolvedFilename, options.cwd));
             }
         } else {
+
+            // if there's a cache, populate it
+            if ("_rulesCache" in this) {
```

Comment:
Will this map contain rules from plugins? There's an outstanding bug in https://github.com/eslint/eslint/issues/10838 where the rules in `CLIEngine` aren't available until the first time the `CLIEngine` instance runs on text, because it doesn't load plugins from a config until it meeds to use that config to lint a file.

This might not be a problem here if the config is fully resolved before the rules are obtained, but it would be good to double-check that this is the case.

Reply:
Yes. This call to `getRules()` is executed every time `executeOnFiles()` is called just before the linting actually happens. At that point, all plugins have been loaded and any calls to `linter.defineRule()` have been called. Note that `"_rulesCache" in this` only checks for the existence of the property (as a shorthand for having valid `fixTypes`) and does not check the property's value.